src/cascadia/TerminalApp/TerminalSettings.cpp

```diff
@@ -6,8 +6,202 @@
+        }
+        if (profile.CursorColor())
+        {
+            _CursorColor = til::color{ profile.CursorColor().Value() };
+        }
+
+        _ScrollState = profile.ScrollState();
+
+        if (!profile.BackgroundImagePath().empty())
+        {
+            _BackgroundImage = profile.GetExpandedBackgroundImagePath();
+        }
+
+        _BackgroundImageOpacity = profile.BackgroundImageOpacity();
+        _BackgroundImageStretchMode = profile.BackgroundImageStretchMode();
+
+        _BackgroundImageHorizontalAlignment = profile.BackgroundImageHorizontalAlignment();
+        _BackgroundImageVerticalAlignment = profile.BackgroundImageVerticalAlignment();
+
+        _RetroTerminalEffect = profile.RetroTerminalEffect();
+
+        _AntialiasingMode = profile.AntialiasingMode();
+
+        if (profile.TabColor())
+        {
+            const til::color colorRef{ profile.TabColor().Value() };
+            _TabColor = static_cast<uint32_t>(colorRef);
+        }
+    }
+
+    // Method Description:
+    // - Applies appropriate settings from the globals into the given TerminalSettings.
+    // Arguments:
+    // - settings: a TerminalSettings object to add global property values to.
+    // Return Value:
+    // - <none>
+    void TerminalSettings::_ApplyGlobalSettings(const TerminalApp::GlobalAppSettings& globalSettings) noexcept
+    {
+        _InitialRows = globalSettings.InitialRows();
+        _InitialCols = globalSettings.InitialCols();
+
+        _WordDelimiters = globalSettings.WordDelimiters();
+        _CopyOnSelect = globalSettings.CopyOnSelect();
+        _ForceFullRepaintRendering = globalSettings.ForceFullRepaintRendering();
+        _SoftwareRendering = globalSettings.SoftwareRendering();
+        _ForceVTInput = globalSettings.ForceVTInput();
+    }
+
+    // Method Description:
+    // - Apply our values to the given TerminalSettings object. Sets the foreground,
+    //      background, and color table of the settings object.
+    // Arguments:
+    // - terminalSettings: the object to apply our settings to.
+    // Return Value:
+    // - <none>
+    void TerminalSettings::_ApplyColorScheme(const TerminalApp::ColorScheme& scheme)
+    {
+        _DefaultForeground = til::color{ scheme.Foreground() };
+        _DefaultBackground = til::color{ scheme.Background() };
+        _SelectionBackground = til::color{ scheme.SelectionBackground() };
+        _CursorColor = til::color{ scheme.CursorColor() };
+
+        const auto table = scheme.Table();
+        auto const tableCount = gsl::narrow_cast<int>(table.size());
+        for (int i = 0; i < tableCount; i++)
+        {
+            SetColorTableEntry(i, til::color{ table[i] });
+        }
+    }
+
+    bool TerminalSettings::ApplyColorScheme(const hstring& scheme, const Windows::Foundation::Collections::IMapView<hstring, TerminalApp::ColorScheme>& schemes)
+    {
+        if (const auto found{ schemes.TryLookup(scheme) })
+        {
+            _ApplyColorScheme(found);

```

Comment:
Curious why the code is split into the private `_ApplyColorScheme` and the public `ApplyColorScheme`?

Reply:
We use the public one to be able to preview a ColorScheme via the CommandPalette (see `_HandleSetColorScheme` in `AppActionHandlers`). I _could_ do a minor refactor and do the following:
- `TerminalSettings`: just make `ApplyColorScheme(ColorScheme)` public (delete the other one)
- `AppActionHandlers`:
   - lookup `ColorScheme` here
   - apply it, if one was found

You know what, I'll just do that now. It'll make this look a bit cleaner :)